core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -38,10 +39,12 @@ import org.apache.spark.util.kvstore.KVStore
  */
 private[spark] abstract class LiveEntity {
 
-  var lastWriteTime = 0L
+  var lastWriteTime = -1L
 
-  def write(store: KVStore, now: Long): Unit = {
-    store.write(doUpdate())
+  def write(store: ElementTrackingStore, now: Long, checkTriggers: Boolean = false): Unit = {
+    // Always check triggers on the first write, since adding an element to the store may
+    // cause the maximum count for the element type to be exceeded.
```

Comment:
hmm, for the first write, how can it trigger maximum count?

Reply:
Multiple jobs, multiple stages, multiple tasks, etc, etc, etc.